airbyte-integrations/connectors/source-okta/source_okta/components.py

```diff
@@ -90,15 +95,15 @@ def token(self) -> str:
         jwt_payload = {
             "iss": client_id,
             "sub": client_id,
-            "aud": f"https://{domain}.okta.com/oauth2/v1/token",
+            "aud": f"https://{sub_domain}.{domain}/oauth2/v1/token",
```

Comment:
I see what you did there, but effectively you're changing what `domain` config key is doing. I agree that the naming that you proposed works better. I also am not sure how config migrations would work.

Instead of effectively renaming a key, what you could do instead is an additive config change with a default value for the new setting:
- Use `config['domain']` for the subdomain, as before
- Make a new thing called `config['environment_domain']` that can either be a string with default value of `okta.com`, or a `oneOf` that covers `okta.com`, `oktapreview.com` etc.

That way, all connections would be able to update to this version without breaking their syncs.